internal.nix

```diff
@@ -76,20 +76,42 @@ rec {
             inherit rev ref;
             allRefs = true;
           };
+
+      # If sourceMods.${name} is set, call it with some information about this
```

Comment:
In the `API.md` I at documented it as only a `version` attribute for now, leaving the possibility for more in the future open. The `version` however is a bit ambiguous, because for `github:` dependencies it could be a git hash or any other git reference. Maybe it would be better to let users distinguish between `github:` and url sources, e.g. by setting `sourceType = "github" / "url"`

Reply:
I changed this a bit, now these attributes are passed:
- `version` for HTTP tarball sources
- `github = { org, repo, rev, ref }` for GitHub sources

Considering that sources for a specific package should typically all be of the same source type, this should also work fairly well.

I also thought a lot about providing a proper version in the GitHub case:
- It's possible to get the version at build time via the `.version` field in the `package.json` file. This would be a uniform way of getting a correct version
- However we lose the ability to decide on which patches to include at eval time
- In order to still be able to limit patches to specific versions, it would be best to have some way of specifying [NPM version ranges](https://github.com/npm/node-semver#ranges) for each patch and calling [runtime checks](https://github.com/npm/node-semver#ranges-1) to figure out whether to include them.
- I deemed this approach too complicated though, because it would have to involve coming up with our own custom scheme of declaring patches and patch commands, entirely separately from the current nixpkgs way.
- The `version` and `github` based version decisions should be good enough for most use cases for now.